client/asset/dcr/wallet.go

```diff
@@ -69,19 +67,14 @@ type Wallet interface {
 	// notification is unimplemented, monitorBlocks should be used to track
 	// tip changes.
 	NotifyOnTipChange(ctx context.Context, cb TipChangeCallback) bool
-	// AccountOwnsAddress checks if the provided address belongs to the
-	// specified account.
-	AccountOwnsAddress(ctx context.Context, account, address string) (bool, error)
-	// AccountBalance returns the balance breakdown for the specified account.
-	AccountBalance(ctx context.Context, account string, confirms int32) (*walletjson.GetAccountBalanceResult, error)
-	// LockedOutputs fetches locked outputs for the specified account.
-	LockedOutputs(ctx context.Context, account string) ([]chainjson.TransactionInput, error)
-	// EstimateSmartFeeRate returns a smart feerate estimate.
-	EstimateSmartFeeRate(ctx context.Context, confTarget int64, mode chainjson.EstimateSmartFeeMode) (float64, error)
-	// Unspents fetches unspent outputs for the specified account.
-	Unspents(ctx context.Context, account string) ([]walletjson.ListUnspentResult, error)
-	// GetChangeAddress returns a change address from the specified account.
-	GetChangeAddress(ctx context.Context, account string) (stdaddr.Address, error)
+	// OwnsAddress checks if the provided address belongs to the Wallet.
+	OwnsAddress(ctx context.Context, addr stdaddr.Address) (bool, error)
+	// Balance returns the balance breakdown for the Wallet.
+	Balance(ctx context.Context, confirms int32) (*walletjson.GetAccountBalanceResult, error)
```

Comment:
When the `ExchangeWallet` calls `wallet.Balance`, it expects the balance for a specific account not the entire wallet. This seems to give a different meaning to `Wallet`. `wallet.AccountBalance` makes more sense to me.

Also, re mixed account support, balance reporting may have to deal with more than one account. With this change, implementations have to somehow understand how to properly report balance for mixed and unmixed accounts; and it won't be as simple as summing the sub balances for each account. This can be made simpler by letting the ExchangeWallet handle proper balance reporting while the `Wallet` does a basic wallet work - return the simple balance for a specific account.